src/GafferSceneUI/Inspector.cpp

```diff
@@ -301,6 +310,303 @@ void Inspector::editScopeInputChanged( const Gaffer::Plug *plug )
+	}
+}
+
+ConstRunTimeTypedPtr Inspector::HistoryPath::property( const InternedString &name, const Canceller *canceller) const
+{
+
+	if( isLeaf() && name == g_namePropertyName )
+	{
+		// Remove the plug name from the end.
+		using Tokenizer = boost::tokenizer<boost::char_separator<char>>;
+
+		std::string nodeString;
+
+		ConstStringDataPtr nameData = runTimeCast<const StringData>( Path::property( g_namePropertyName, canceller ) );
+		std::string name = nameData->readable();
+
+		Tokenizer tokens( name, boost::char_separator<char>( "." ) );
+
+		for( Tokenizer::const_iterator it = tokens.begin(), eIt = tokens.end(); it != eIt; ++it )
+		{
+			Tokenizer::const_iterator next = it;
+			next++;
+			if( next == tokens.end() )
+			{
+				continue;
+			}
+			if( nodeString.size() )
+			{
+				nodeString += ".";
+			}
+			nodeString += *it;
+		}
+
+		return new StringData( nodeString );
+	}
+
+	if(
+		isLeaf() && (
+			name == g_valuePropertyName ||
+			name == g_operationPropertyName ||
+			name == g_acquiredEditPropertyName ||
+			name == g_editWarningPropertyName
+		)
+	)
+	{
+		const std::string pathName = names()[0].string();
+
+		Context::Scope scope( m_context.get() );
+
+		SceneAlgo::History::ConstPtr history = m_inspector->history();
+		const SceneAlgo::History *currentHistory = history.get();
+
+		ScenePlug *targetPlug = m_scriptNode.get()->descendant<ScenePlug>( pathName );
+
+		assert( targetPlug );
+
+		while( true )
+		{
+			std::string editWarning;
+
+			Context::Scope currentScope( currentHistory->context.get() );
+
+			if( ValuePlugPtr immediateSource = m_inspector->source( currentHistory, editWarning ) )
+			{
+				if( currentHistory->scene == targetPlug )
+				{
+					ValuePlug *source = static_cast<ValuePlug *>( spreadsheetAwareSource( immediateSource.get() ) );
+					const std::string nodeName = currentHistory->scene->relativeName( m_scriptNode.get() );
+
+					if( name == g_valuePropertyName )
+					{
+						return runTimeCast<const IECore::Data>( m_inspector->value( currentHistory ) );
+					}
+					else if( name == g_operationPropertyName )
+					{
+						if( auto tweakPlug = runTimeCast<const TweakPlug>( source ) )
+						{
+							return new IntData( tweakPlug->modePlug()->getValue() );
+						}
+						return new IntData( TweakPlug::Mode::Create );
+					}
+					else if( name == g_acquiredEditPropertyName )
+					{
+						return source;
+					}
+					else if( name == g_editWarningPropertyName )
+					{
+						return new StringData( editWarning );
+					}
+				}
+			}
+
+			if( currentHistory->predecessors.size() == 0 )
+			{
+				break;
+			}
+			currentHistory = currentHistory->predecessors[0].get();
+		}
+	}
+
+	return Path::property( name );
+}
+
+bool Inspector::HistoryPath::isLeaf( const Canceller *canceller) const
+{
+	return names().size() > 0;
+}
+
+PathPtr Inspector::HistoryPath::copy() const
+{
+	return new Inspector::HistoryPath( m_inspector, m_context, root(), const_cast<PathFilter *>( getFilter() ) );
```

Comment:
Why are we constructing with `root()` here, rather than `string()`? Won't that mean that copying a leaf path will result in a non-leaf path?

Reply:
Fixed in 3ae4fd49def6f524869da06a3badad45cf6c4c0d